lib/travis/logs/services/upsert_log.rb

```diff
@@ -0,0 +1,37 @@
+module Travis
+  module Logs
+    module Services
+      class UpsertLog
+        def initialize(database: nil)
+          @database = database || Travis::Logs.database_connection
+        end
+
+        attr_reader :database
+        private :database
+
+        def run(job_id: nil, content: '', removed_by: nil, clear: false)
+          job_id = Integer(job_id)
+          log_id = find_or_create_log(job_id)
+
+          content = content.to_s
+          content = nil if content.empty?
```

Comment:
Are there any cases where `nil` content and `""` content differs? Trying to figure out if it would ever be useful to be able to set the content in the DB to an empty string instead of NULL.